### ISSUE TITLE
Placement by tapping in Geotrace and Geoshape widget used with Google maps doesn’t add points correctly between two exsting points

### ISSUE THREAD

**Issue by dbemke** [getodk/collect#5226]
**Software and hardware versions** 

Collect v2021.2.0
Android probably all

**Problem description**

When users chooses Google maps and goes to Geotrace widget ,uses placement by tapping and creates a trace with at least 3 points creating V-shape, and tries to add points between the last and the first point already added (if they are close and not connected), new points on the map are not created but the number of points entered changes.
In case of Geoshape widget the problem occurs when users added at least two points and tries to add a point close to an existing point.

https://user-images.githubusercontent.com/56479916/181771238-3faa1eec-1318-430d-b209-ab426317ff90.mp4

https://user-images.githubusercontent.com/56479916/181771302-e1e0f7f4-f4d3-4b34-8fef-5222a0ec494c.mp4

**Steps to reproduce the problem**

1. Select Google maps as a source of maps.
2. Go to All widgets form.
3. Go to Geotrace widget.
4. Click „Start Geotrace”.
5. Click add points button and choose „placement by tapping”.
6. Click start.
7. Add at least 3 points creating V-shape.
8. Add points between the last and the first point (close to an existing point).

**Expected behavior**

User is able to add points between 2 points that are close to each other. Number of points entered changes and there are markers created.

**Other information**

I’ve also check on Android 8.1 and 11 Collect v2022.2.3 , Collect v2022.2.0 and Collect v2022.3 Beta 2 and it doesn’t work as expected. I haven’t checked earlier versions than Collect v2021.2.0.

**Comment by seadowg** [comment]
@dbemke is this only a problem in Google Maps?

> I’ve also check on Android 8.1 and 11 Collect v2022.2.3 , Collect v2022.2.0 and Collect v2022.3 Beta 2 and it doesn’t work as expected. I haven’t checked earlier versions than Collect v2021.2.0.

Does that mean you have checked `v2021.2.0` and the issue is present? Or do you mean `v2022.2.0` as that's what you mention in the first sentence.

**Comment by dbemke** [comment]
@seadowg I've just checked again and it's only in Google Maps. I checked v2021.2.0  and the issue is present. I also checked other mentioned versions and the issue is present.

**Comment by dbemke** [comment]
Similar thing happens in Google maps when a user taps near a point in Select one from map form. It is possible to tap near the point and the point will be selected.

https://user-images.githubusercontent.com/56479916/198229971-1054037d-bdcc-41ef-84ab-ea14038a691f.mp4

**Steps to reproduce the problem:**

1. Go to the demo project.
2. Go to Select one from map form.
3. Tap "Select place".
4. Tap near a point on the map.

**Comment by srujner** [comment]
@seadowg 
I've reproduced today this old issue on Geotrace and Google maps and I noticed one more things that looks like has not been mentioned before. 
Once the last point connects to the first one they have the same geometry 

![Image](https://github.com/user-attachments/assets/3d708690-9c43-4b30-818d-4c2d7ceb34b1)

**Comment by seadowg** [comment]
@srujner definitely looks like it's "snapping" points? Maybe Google Maps snaps click events to nearby features?

**Comment by lognaturel** [comment]
> Google Maps snaps click events to nearby features

Yes, I recently discovered this as well: https://getodk.slack.com/archives/C03QX56DD8C/p1764874521515649
